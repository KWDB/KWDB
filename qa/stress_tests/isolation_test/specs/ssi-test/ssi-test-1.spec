# select behavior (without FOR UPDATE)
setup
{
 CREATE TABLE IF NOT EXISTS kv(k int PRIMARY key, v int);
 delete from kv;
 INSERT INTO kv VALUES (1,5);
}

teardown
{
 DROP TABLE kv;
}

session s1
step s1   { BEGIN transaction isolation level SERIALIZABLE; }
step s1r1 { SELECT * FROM kv; }
step s1r2 { SELECT * FROM kv; }
step s1w1 { INSERT INTO kv VALUES(3,7); }
step s1r3 { SELECT * FROM kv; }
step s1r4 { SELECT * FROM kv; }
step s1c  { COMMIT; }

session s2
step s2   { BEGIN transaction isolation level SERIALIZABLE; }
step s2w1 { INSERT INTO kv VALUES(2,6); }
step s2c  { COMMIT; }

permutation s1 s2 s1r1 s2w1 s1r2 s1w1 s1r3 s2c s1r4 s1c