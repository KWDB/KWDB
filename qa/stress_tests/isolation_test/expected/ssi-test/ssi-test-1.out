Parsed test spec with 2 sessions

starting permutation: s1 s2 s1r1 s2w1 s1r2 s1w1 s1r3 s2c s1r4 s1c
step s1: BEGIN transaction isolation level SERIALIZABLE;  finished.
step s2: BEGIN transaction isolation level SERIALIZABLE;  finished.
step s1r1: SELECT * FROM kv;  finished.
k|v
-+-
1|5
(1 row)

step s2w1: INSERT INTO kv VALUES(2,6);  finished.
step s1r2: SELECT * FROM kv;  finished.
k|v
-+-
1|5
(1 row)

step s1w1: INSERT INTO kv VALUES(3,7);  finished.
step s1r3: SELECT * FROM kv;  finished.
k|v
-+-
1|5
3|7
(2 rows)

step s2c: COMMIT;  finished.
step s1r4: SELECT * FROM kv;  finished.
k|v
-+-
1|5
3|7
(2 rows)

step s1c: COMMIT;  finished.
